### PR TITLE
doc: update board porting documentation page

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR updates the boards porting documentation after #15477 and #15475 were merged since now it's not necessary to include `<programmer>.inc.mk` and `serial.inc.mk` at the board level.

### Testing procedure

Read the documentation

### Issues/PRs references

Doc needs an update after #15477 and #15475 are merged. Depends on #15477

### CODE REVIEW

File: doc/doxygen/src/porting-boards.md [hunk 1]
```diff
@@ -135,11 +135,8 @@ e.g.:
 PORT_LINUX ?= /dev/ttyUSB0
 PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbserial*)))
```
Comment (fjmolinas) [inline doc/doxygen/src/porting-boards.md:136]: I would add a comment that this is only if you need to override defaults

**Comment by fjmolinas** [comment]
I think you can only leave the `doc` commits on this one, will avoid a rebase.

**Comment by fjmolinas** [comment]
please rebase

File: doc/doxygen/src/porting-boards.md (hunk 1 above)
Comment (aabadie) [reply doc/doxygen/src/porting-boards.md:149]: I extended the documentation.

**Review by fjmolinas — APPROVED** [review]
ACK

**Pull request merged**